### PR TITLE
Create initial person, student & teacher classes

### PR DESCRIPTION
**Pull request by rominana**
### Project requirements 

- Create class Person with the following:

Instance vars: @id, @name, and @age.
Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
Getters for @id, @name, and @age.
Setters for @name and @age.
Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- Create class Student with the following:

Inherits from Person.
Constructor extends parent's constructor by adding @classroom and a parameter for it.
Method play_hooky that returns "¯\(ツ)/¯".

- Create class Teacher with the following:

Inherits from Person.
Constructor extends parent's constructor by adding @specialization and a parameter for it.
Override can_use_services? so it always returns true.
Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
+    @id = Random.rand(1..100)
+  end
+
+  def propper_age?
+    @age >= 18
+  end
```
Comment (ShahierNashaat) [inline person.rb:14]: - [ ] Kindly make this method private as per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Private%20method%20is_of_age%3F%20that%20returns%20true%20if%20%40age%20is%20greater%20or%20equal%20to%2018%20and%20false%20otherwise.).

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @rominana, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rominana** [comment]
All the suggested changes have been applied :smile: 

BUT 
When I made the method private it doesn't pass the linters: 
```
  def propper_age?
    if @age >= 18
        return true
    else 
        return false
    end
  end
```

When I apply `rubocop -a`, my code changes and gets back to the original code
```
  def propper_age?
    @age >= 18
  end

```
what can I do in this case? 
please take this into consideration, **it's a linter problem.**

**Comment by m05e5** [comment]
Hi @rominana the code you wrote at first was right 
 `def propper_age?`
 `  @age >= 18`
  `end`
to make the method private you just have add the `private` key word on top of that method and all the methods below will be private, or you call specify that method as private with `private :propper_age`

**Comment by m05e5** [comment]
so either 
```
private


def propper_age?
    @age >= 18
  end
```

or 

```
def propper_age?
    @age >= 18
  end

private  :propper_age
```

**Comment by rominana** [comment]
> Hi @rominana the code you wrote at first was right `def propper_age?` ` @age >= 18` `end` to make the method private you just have add the `private` key word on top of that method and all the methods below will be private, or you call specify that method as private with `private :propper_age`

**Comment by m05e5** [comment]
Please Cant review a closed PR can you reopen it?

**Comment by rominana** [comment]
Yes, I accidentally close it! So sorry @m05e5! Many thanks for your help

**Review by m05e5 — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @rominana   🔥 

Good job on implementing the requested changes. You smashed it. Well Done.
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

goodjob


Now you can merge this branch 🚀 

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

_m05e5_

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**